File: modules/user/src/main/scala/shared/TopicUtils.scala
package shared
import io.simplesource.saga.shared.topics.{TopicConfigBuilder, TopicNamer}

object TopicUtils {

  def buildSteps(prefix: String, baseName: String): shared.topics.TopicConfigBuilder.BuildSteps =
    builder =>
      builder
        .withTopicNamer(shared.topics.TopicNamer.forPrefix(prefix, baseName))
        .withDefaultConfig(constants.partitions, constants.replication, constants.retentionDays)

  def buildStepsJ(prefix: String, baseName: String): TopicConfigBuilder.BuildSteps =
    builder =>
      builder
        .withTopicNamer(TopicNamer.forPrefix(prefix, baseName))
        .withDefaultConfig(constants.partitions, constants.replication, constants.retentionDays)
}

File: simplesaga-testutils/src/main/java/io/simplesource/saga/testutils/TopicUtils.java
package io.simplesource.saga.testutils;

import io.simplesource.saga.shared.topics.TopicConfigBuilder;
import io.simplesource.saga.shared.topics.TopicNamer;

public class TopicUtils {
    public static TopicConfigBuilder.BuildSteps buildSteps(String prefix, String baseName) {
        return builder -> builder.withTopicNamer(TopicNamer.forPrefix(prefix, baseName))
                .withDefaultConfig(Constants.PARTITIONS, Constants.REPLICATION, Constants.RETENTION_DAYS);
    }
}
